oxide/lib.go

```diff
@@ -118,6 +158,89 @@ func defaultUserAgent() string {
 	return fmt.Sprintf("oxide.go/%s", version)
 }
 
+// getProfile determines the path of the user's credentials file
+// and returns the host and token for the requested profile.
+func getProfile(cfg Config) (*authCredentials, error) {
+	if cfg.Host != "" || cfg.Token != "" {
+		return nil, errors.New("cannot authenticate with both a profile and host/token")
+	}
```

Comment:
Let's move this validation to the `NewClient` function. We don't want to call the `getProfile` function if the conditions haven't been met ye. It also helps readability of the code :)

Reply:
Agree that is cleaner, done in 0fca9c247f7688d4c2879fa986ad00a0bc8f8a69